Ginger/GingerCoreNET/ActionsLib/UI/Web/ActAccessibilityTesting.cs

```diff
@@ -402,5 +408,380 @@ private void RuleData_PropertyChanged(object? sender, PropertyChangedEventArgs e
+                htmlAndSelector.SetAttributeValue("class", "wrapTwo");
+                htmlAndSelectorWrapper.AppendChild(htmlAndSelector);
+
+                if (allCheckResults.Length > 0 || noneCheckResults.Length > 0)
+                {
+                    FixAllIssues(doc, htmlAndSelectorWrapper, allCheckResults, noneCheckResults);
+                }
+
+                if (anyCheckResults.Length > 0)
+                {
+                    FixAnyIssues(doc, htmlAndSelectorWrapper, anyCheckResults);
+                }
+            }
+        }
+
+        private static void FixAllIssues(HtmlDocument doc, HtmlNode htmlAndSelectorWrapper, AxeResultCheck[] allCheckResults, AxeResultCheck[] noneCheckResults)
+        {
+            HtmlNode htmlAndSelector;
+
+            htmlAndSelector = doc.CreateElement("p");
+            htmlAndSelector.SetAttributeValue("class", "wrapOne");
+            StringBuilder content = new StringBuilder();
+
+            content.AppendLine("Fix all of the following issues:");
+            content.AppendLine("<ul>");
+
+            foreach (var checkResult in allCheckResults)
+            {
+                content.AppendLine($"<li>{HttpUtility.HtmlEncode(checkResult.Impact.ToUpper())}: {HttpUtility.HtmlEncode(checkResult.Message)}</li>");
+            }
+
+            foreach (var checkResult in noneCheckResults)
+            {
+                content.AppendLine($"<li>{HttpUtility.HtmlEncode(checkResult.Impact.ToUpper())}: {HttpUtility.HtmlEncode(checkResult.Message)}</li>");
+            }
+
+            content.AppendLine("</ul>");
+            htmlAndSelector.InnerHtml = content.ToString();
+            htmlAndSelectorWrapper.AppendChild(htmlAndSelector);
+        }
+
+        private static void FixAnyIssues(HtmlDocument doc, HtmlNode htmlAndSelectorWrapper, AxeResultCheck[] anyCheckResults)
+        {
+            StringBuilder content = new StringBuilder();
+
+            HtmlNode htmlAndSelector = doc.CreateElement("p");
+            htmlAndSelector.SetAttributeValue("class", "wrapOne");
+            content.AppendLine("Fix at least one of the following issues:");
+            content.AppendLine("<ul>");
+
+            foreach (var checkResult in anyCheckResults)
+            {
+                content.AppendLine($"<li>{HttpUtility.HtmlEncode(checkResult.Impact.ToUpper())}: {HttpUtility.HtmlEncode(checkResult.Message)}</li>");
+            }
+
+            content.AppendLine("</ul>");
+            htmlAndSelector.InnerHtml = content.ToString();
+            htmlAndSelectorWrapper.AppendChild(htmlAndSelector);
+        }
+
+        private static string GetCss(ISearchContext context)
+        {
+            return EmbeddedResourceProvider.ReadEmbeddedFile("htmlReporter.css").Replace("url('", $"url('{GetDataImageString(context)}");
+        }
+
+        private static string GetDataImageString(ISearchContext context)
+        {
+            ITakesScreenshot newScreen = (ITakesScreenshot)context;
```

Comment:
Please check if we can use Using to avoid memory leaks

Reply:
not abled to use using here